backend/app/services/rag/pipeline/llamaindex.py

```
@@ -0,0 +1,248 @@
+            chunk_size: Maximum chunk size in characters
+            chunk_overlap: Number of characters to overlap between chunks
+        """
+        super().__init__(chunk_size, chunk_overlap)
+        self.file_extension = file_extension.lower()
+
+    @classmethod
+    def get_supported_extensions(cls) -> set:
+        """Get supported file extensions."""
+        return cls.SUPPORTED_EXTENSIONS
+
+    def read(self, binary_data: bytes, file_extension: str) -> bytes:
```

Ruff (0.15.0)

[warning] Unused method argument: `file_extension`

(ARG002)